browser/src/libs/github/util.tsx

```
@@ -182,29 +182,44 @@ function getDiffResolvedRevFromPageSource(pageSource: string, isPullRequest: boo
- * present on GitHub blob pages.
+ * Returns the file path for the current page. Must be on a blob page.
+ *
+ * Implementation details:
+ *
+ * - This scrapes the file path from the permalink on GitHub blob pages:
+ *
+ *     <a class="d-none js-permalink-shortcut" data-hotkey="y" href="/gorilla/mux/blob/ed099d42384823742bba0bf9a72b53b55c9e2e38/mux.go">Permalink</a>
+ *
+ * - We can't get the file path from the URL because the branch name can contain
+ *   slashes which make the boundary between the branch name and file path
+ *   ambiguous. For example:
+ *
+ *     https://github.com/sourcegraph/sourcegraph/blob/bext/release/cmd/frontend/internal/session/session.go
+ *
+ * TODO ideally, this should only scrape the code view itself.
  */
-export function getBranchName(): string {
-    const branchSelectMenu = document.querySelector('.branch-select-menu') as HTMLElement
-    if (!branchSelectMenu) {
-        throw new Error('Could not find .branch-select-menu')
-    }
-    const cssTruncateTarget = branchSelectMenu.querySelector('span.css-truncate-target') as HTMLElement
-    if (!cssTruncateTarget) {
-        throw new Error('Could not find span.css-truncate-target')
-    }
-    if (!cssTruncateTarget.innerText.endsWith('…')) {
-        // The branch name is not truncated
-        return cssTruncateTarget.innerText
-    }
-    // When the branch name is truncated, it is stored in full
-    // in the select menu button's title attribute.
-    const selectButton = cssTruncateTarget.closest('.select-menu-button') as HTMLElement
-    if (!selectButton) {
-        throw new Error('Could not find .select-menu-button')
-    }
-    return selectButton.title
+export function getFilePath(): string | null {
+    const permalink = document.querySelector('.js-permalink-shortcut')
+    if (!permalink) {
+        console.error(`Unable to determine the file path because no .js-permalink-shortcut element was found.`)
+        return null
```

Imo it's better to throw an Error than to return `null` - an Error should be thrown if a function cannot fulfill its contract, which is the case here. The caller then probably also cannot fulfill _its_ contract, so it's useful to be able to let the exception bubble up (vs a Go style checking for null at every level) and persist the context why it failed vs just a global log message

---

Agreed, switched to throwing an error.